README.md

```diff
@@ -240,6 +240,22 @@ To send the transaction, you can call the `writeAsync` function returned by the
 
 This example sends a transaction to the `YourContract` smart contract to call the `setGreeting` function with the arguments passed in `args`. The `writeAsync` function sends the transaction to the smart contract, and the `isLoading` and `isMining` properties indicate whether the transaction is currently being processed by the network.
 
+If you have any state, you can just pass it into the `args`, and it will automatically update when you call `writeAsync()`. Example:
```

Comment:
The wording is a bit confusing. It's not so much the args that are updated when calling the `writeAsync`, it's the writeAsync` itself that get's updated when a state that is used by `useScaffoldContractWrite` has changed due to react rerender.

So this mechanism is just how react works, and isn't specific to this code.

Reply:
agreed, since its more about how react works. Would suggest to keep it the way it is here instead of adding the clarification. As @carletex mentioned we can get into more details with the recipes.